pkg/api/cluster_types.go

```diff
@@ -516,3 +520,23 @@ func (cluster *Cluster) GetSupportedInstanceTypes() []string {
 func (cluster *Cluster) GetRawSupportedInstanceTypes() string {
 	return cluster.SupportedInstanceType
 }
+
+// SetPrewarmingStatusInfo sets the prewarming status info per instance type into a json object that can be persisted in
+// the database
+func (cluster *Cluster) SetPrewarmingStatusInfo(prewarmingStatusInfo map[string]map[string]int) {
```

Comment:
I am thinking that we it might be worth investing some time trying to make this more typesafe compared to previous implementations of similar data like DynamicCapacityInfo or AvailableStrimziVersions:

We are exposing that prewarmingStatusInfo as a map[string]map[string]int. It is unavoidable at JSON/db level the way we are doing it (at serialization time). However, at code level I think we can be safer and more flexible:

Make map[string]map[string]int its own type. For that we have two options:
```
type prewarmingStatusIndex map[string]map[string]int // or prewarmingStatusInfoIndex
```

or
```
type prewarmingStatusIndex struct { // or prewarmingStatusInfoIndex
  index map[string]map[string]int
}  
```
I tend to prefer the later. The difference is that with the first approach we indirectly coupling the implementation to a map, whereas with the second approach that is totally hidden and thus transparent to change. With the first approach we could directly convert it to a map and use the typical Go map operators but I think that is even counterproductive as again it ties the type to map-specific semantics.

From that, we can even go further. the second map[string]int can be its own type too, and the same approach would apply to it.

Additional useful methods and potentially safer could be then implemented on top of the new types too.

Reply:
We can work on this together if needed

Reply:
I was leaning toward the second approach for this e.g having a 
```go
type PrewarmingInfo struct { // or PrewarmingInfo
  statusCountPerInstanceType map[string]CountPerStatus
} 

type CountPerStatus map[string]int
```
where the whole PrewarmingInfo struct is marshalled/unmarshalled. 

I had a mental blocker about adding the type now as I cant see any other field that we'll like to add in `PrewarmingInfo` struct. Can you think of any?

Reply:
PrewarmingInfoIndex

Reply:
On the other hand, `CountPerStatus` seems very generic. We should be more specific if possible

Reply:
Thanks @miguelsorianod for the review. This is no longer something that we need to pursue since this logic has been removed from the PR as of https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/commit/6e4f7374f2ff2f1d48d21d0581c5879b9e712095